tnt-service/src/main/java/org/dcsa/tnt/service/mapping/EventMapper.java

```diff
@@ -1,31 +1,90 @@
 package org.dcsa.tnt.service.mapping;
 
-import org.dcsa.tnt.domain.valueobjects.DomainEvent;
-import org.dcsa.tnt.domain.valueobjects.EquipmentEvent;
-import org.dcsa.tnt.domain.valueobjects.ShipmentEvent;
-import org.dcsa.tnt.domain.valueobjects.TransportEvent;
-import org.dcsa.tnt.transferobjects.EquipmentEventPayloadTO;
-import org.dcsa.tnt.transferobjects.EventMetadataTO;
-import org.dcsa.tnt.transferobjects.EventTO;
-import org.dcsa.tnt.transferobjects.ShipmentEventPayloadTO;
-import org.dcsa.tnt.transferobjects.TransportEventPayloadTO;
+import org.dcsa.skernel.infrastructure.transferobject.AddressTO;
+import org.dcsa.tnt.domain.valueobjects.*;
+import org.dcsa.tnt.transferobjects.*;
+import org.dcsa.tnt.transferobjects.enums.LocationType;
 import org.mapstruct.Mapper;
 
 @Mapper(componentModel = "spring")
 public abstract class EventMapper {
   public EventTO toDTO(DomainEvent event) {
     return EventTO.builder()
-      .metadata(toMetadataTO(event))
-      .payload(switch (event.getEventType()) {
-        case EQUIPMENT -> toEquipmentEventPayloadTO((EquipmentEvent) event);
-        case SHIPMENT -> toShipmentEventPayloadTO((ShipmentEvent) event);
-        case TRANSPORT -> toTransportEventPayloadTO((TransportEvent) event);
-      })
-      .build();
+        .metadata(toMetadataTO(event))
+        .payload(
+            switch (event.getEventType()) {
+              case EQUIPMENT -> toEquipmentEventPayloadTO((EquipmentEvent) event);
+              case SHIPMENT -> toShipmentEventPayloadTO((ShipmentEvent) event);
+              case TRANSPORT -> toTransportEventPayloadTO((TransportEvent) event);
+            })
+        .build();
   }
 
   protected abstract EventMetadataTO toMetadataTO(DomainEvent event);
+
   protected abstract EquipmentEventPayloadTO toEquipmentEventPayloadTO(EquipmentEvent event);
+
   protected abstract ShipmentEventPayloadTO toShipmentEventPayloadTO(ShipmentEvent event);
+
   protected abstract TransportEventPayloadTO toTransportEventPayloadTO(TransportEvent event);
+
+  protected LocationTO locationToLocationTO(Location location) {
+
+    LocationTO loc = null;
+
+    if (null != location) {
+
+      if (null != location.facilityCode()) {
+        loc =
+            FacilityLocationTO.builder()
+                .locationName(location.locationName())
+                .locationType(LocationType.FACI)
```

Comment:
Why not use the `locationType` to switch on? The purpose of the `locationType` is to be used as a discriminator..